### PR TITLE
Adf Percent Change Plot Panels

### PR DESCRIPTION
**Pull request by shawnusaf**
Pull request adds/addresses the following:

1.  lat/lon/polar/zonal/meridional plots have a 4th panel that plots out the % difference between the base and test cases.
2.  Many of the chemistry variables are added into the adf_variable_defaults.yaml file.  

Item 1 addresses https://github.com/NCAR/ADF/issues/331.

-Shawn

### CODE REVIEW

**Comment by shawnusaf** [comment]
Made a few additional fixes to plotting_functions.py to get the checks to pass.

File: lib/adf_variable_defaults.yaml
Comment (justin-richling) [inline lib/adf_variable_defaults.yaml:1]: Thanks for all the added details @shawnusaf! If you could just go through and remove all commented out lines that would be helpful.

File: lib/plotting_functions.py [hunk 1]
```diff
@@ -650,6 +651,16 @@ def make_polar_plot(wks, case_nickname, base_nickname,
         dif = d2 - d1
     else:
         dif = difference
+        
+    if  pctchange is None:
+        #pct = (abs(d2 - d1)/((d2 + d1)/2.0))*100.0
```
Comment (justin-richling) [inline lib/plotting_functions.py:656]: Could you remove this commented out line?

File: lib/plotting_functions.py [hunk 2]
```diff
@@ -668,23 +679,28 @@ def make_polar_plot(wks, case_nickname, base_nickname,
...
+    pct_cyclic, _ = add_cyclic_point(pct, coord=pct.lon)
 
     # -- deal with optional plotting arguments that might provide variable-dependent choices
 
     # determine levels & color normalization:
     minval    = np.min([np.min(d1), np.min(d2)])
     maxval    = np.max([np.max(d1), np.max(d2)])
     absmaxdif = np.max(np.abs(dif))
+    absmaxpct = np.max(np.abs(pct))
+    #print('absmaxpct: ',absmaxpct)
```
Comment (justin-richling) [inline lib/plotting_functions.py:703]: Could you remove this commented out line?

File: lib/plotting_functions.py [hunk 3]
```diff
@@ -714,6 +730,17 @@ def make_polar_plot(wks, case_nickname, base_nickname,
...
+    if "pct_diff_contour_levels" in kwargs:
+        levelspctdiff = kwargs["pct_diff_contour_levels"]  # a list of explicit contour levels
+    elif "pct_diff_contour_range" in kwargs:
+            assert len(kwargs['pct_diff_contour_range']) == 3, "pct_diff_contour_range must have exactly three entries: min, max, step"
+            levelspctdiff = np.arange(*kwargs['pct_diff_contour_range'])
+    else:
+        # set levels for difference plot (with a symmetric color bar):
+        levelspctdiff = np.linspace(-1*absmaxpct, absmaxpct, 15)
+        
+    #print("levelspctdiff=======",levelspctdiff)
```
Comment (justin-richling) [inline lib/plotting_functions.py:743]: Could you remove this commented out line?

File: lib/plotting_functions.py [hunk 4]
```diff
@@ -733,8 +760,13 @@ def make_polar_plot(wks, case_nickname, base_nickname,
...
     #End if
 
-    if max(abs(levelsdiff)) > 10*absmaxdif:
+    if max(np.abs(levelsdiff)) > 10*absmaxdif:
         levelsdiff = np.linspace(-1*absmaxdif, absmaxdif, 12)
+    
+    #don't do this for the percent difference.
+    #if max(np.abs(levelspctdiff)) > 10*absmaxpct:
+    #    levelspctdiff = np.linspace(-1*absmaxpct, absmaxpct, 12)
+
```
Comment (justin-richling) [inline lib/plotting_functions.py:769]: Could you remove these commented out lines?

File: lib/plotting_functions.py [hunk 5]
```diff
@@ -1255,19 +1330,25 @@ def plot_map_and_save(wks, case_nickname, base_nickname,
...
-
-    if "units" in kwargs:
-        ax[1].set_ylabel(f"[{kwargs['units']}]")
-        ax[-1].set_ylabel(f"[{kwargs['units']}]")
-    else:
-        ax[1].set_ylabel(f"{mdlfld.units}")
-        ax[-1].set_ylabel(f"{mdlfld.units}")
+    ax[3].set_title(f"RMSE: {d_rmse:.3f}", fontsize=tiFontSize)
+    ax[3].set_title("$\mathbf{Test} - \mathbf{Baseline}$", loc='left', fontsize=tiFontSize)
+    #ax[2].set_title("$\mathbf{Test} % \mathbf{Diff} \mathbf{Baseline}$", loc='left', fontsize=tiFontSize)
```
Comment (justin-richling) [inline lib/plotting_functions.py:1341]: Could you remove this commented out line?

File: lib/plotting_functions.py [hunk 6]
```diff
@@ -1255,19 +1330,25 @@ def plot_map_and_save(wks, case_nickname, base_nickname,
...
+    ax[2].set_title("Test % Diff Baseline", loc='left', fontsize=tiFontSize,fontweight="bold")
+
+    #if "units" in kwargs:
+        #ax[1].set_ylabel(f"[{kwargs['units']}]")
+        #ax[2].set_ylabel(f"[{kwargs['units']}]")
+        #ax[3].set_ylabel(f"[{kwargs['units']}]")
+    #else:
+    #    ax[1].set_ylabel(f"{mdlfld.units}")
+    #    ax[2].set_ylabel(f"{mdlfld.units}")
+    #    ax[3].set_ylabel(f"{mdlfld.units}")
```
Comment (justin-richling) [inline lib/plotting_functions.py:1351]: Could you remove these commented out lines?

File: lib/plotting_functions.py [hunk 7]
```diff
@@ -1849,27 +1964,40 @@ def prep_contour_plot(adata, bdata, diffdata, **kwargs):
         normdiff = normfunc(vmin=np.min(levelsdiff), vmax=np.max(levelsdiff), vcenter=0.0)
     else:
         normdiff = mpl.colors.Normalize(vmin=np.min(levelsdiff), vmax=np.max(levelsdiff))
+        
+    # color normalization for percent difference
+    if "pct_diff_contour_levels" in kwargs:
+        #normpct = mpl.colors.Normalize(vmin=np.min(levelspctdiff), vmax=np.max(levelspctdiff))
```
Comment (justin-richling) [inline lib/plotting_functions.py:1970]: Could you remove this commented out line?

File: lib/plotting_functions.py [hunk 8]
```diff
@@ -1943,15 +2071,23 @@ def plot_zonal_mean_and_save(wks, case_nickname, base_nickname,
 
         # calculate difference:
         diff = azm - bzm
+        
+        # calculate the percent change
+        #pct = (np.abs(azm - bzm)/((azm + bzm)/2.0))*100.0
```
Comment (justin-richling) [inline lib/plotting_functions.py:2076]: Could you remove this commented out line?

File: lib/plotting_functions.py [hunk 9]
```diff
@@ -1997,8 +2141,16 @@ def plot_zonal_mean_and_save(wks, case_nickname, base_nickname,
         azm = zonal_mean_xr(adata)
         bzm = zonal_mean_xr(bdata)
         diff = azm - bzm
-        fig, ax = plt.subplots(nrows=2)
-        ax = [ax[0],ax[1]]
+        
+        # calculate the percent change
+        #pct = (np.abs(azm - bzm)/((azm + bzm)/2.0))*100.0
```
Comment (justin-richling) [inline lib/plotting_functions.py:2146]: Could you remove this commented out line?

File: lib/plotting_functions.py [hunk 10]
```diff
@@ -2138,6 +2293,13 @@ def plot_meridional_mean_and_save(wks, case_nickname, base_nickname,
         return None
 
     diff = adata - bdata
+    
+    # calculate the percent change
+    #pct = (np.abs(adata - bdata)/((adata + bdata)/2.0))*100.0
```
Comment (justin-richling) [inline lib/plotting_functions.py:2298]: Could you remove this commented out line?

File: lib/plotting_functions.py [hunk 11]
```diff
@@ -2334,6 +2507,13 @@ def square_contour_difference(fld1, fld2, **kwargs):
 
 
     diff = fld1 - fld2
+    
+    #pct = (np.abs(fld1 - fld2)/((fld1 + fld2)/2.0))*100.0
```
Comment (justin-richling) [inline lib/plotting_functions.py:2511]: Could you remove this commented out line?

File: scripts/plotting/global_latlon_map.py [hunk 12]
```diff
@@ -251,11 +252,15 @@ def global_latlon_map(adfobj):
 
                     # difference: each entry should be (lat, lon)
                     dseasons[s] = mseasons[s] - oseasons[s]
+                    
+                    # percent change
+                    #pseasons[s] = (abs(mseasons[s] - oseasons[s])/((mseasons[s] + oseasons[s])/2.0))*100.0
```
Comment (justin-richling) [inline scripts/plotting/global_latlon_map.py:257]: Could you remove this commented out line?

File: scripts/plotting/global_latlon_map.py [hunk 13]
```diff
@@ -291,11 +296,16 @@ def global_latlon_map(adfobj):
 
                         # difference: each entry should be (lat, lon)
                         dseasons[s] = mseasons[s] - oseasons[s]
+                        
+                        # percent change
+                        #pseasons[s] = (abs(mseasons[s] - oseasons[s])/((mseasons[s] + oseasons[s])/2.0))*100.0 #absolute
```
Comment (justin-richling) [inline scripts/plotting/global_latlon_map.py:301]: Could you remove this commented out line?

File: scripts/plotting/polar_map.py [hunk 14]
```diff
@@ -245,6 +246,14 @@ def polar_map(adfobj):
                             # difference: each entry should be (lat, lon)
                             dseasons[s] = mseasons[s] - oseasons[s]
                             dseasons[s].attrs['units'] = mseasons[s].attrs['units']
+                            
+                            # percent change
+                            #pseasons[s] = (abs(mseasons[s] - oseasons[s])/((mseasons[s] + oseasons[s])/2.0))*100.0 #absolute
```
Comment (justin-richling) [inline scripts/plotting/polar_map.py:251]: Could you remove this commented out line?

File: scripts/plotting/polar_map.py [hunk 15]
```diff
@@ -325,6 +335,14 @@ def polar_map(adfobj):
                                 # difference: each entry should be (lat, lon)
                                 dseasons[s] = mseasons[s] - oseasons[s]
                                 dseasons[s].attrs['units'] = mseasons[s].attrs['units']
+                                
+                                # percent change
+                                #pseasons[s] = (abs(mseasons[s] - oseasons[s])/((mseasons[s] + oseasons[s])/2.0))*100.0
```
Comment (justin-richling) [inline scripts/plotting/polar_map.py:340]: Could you remove this commented out line?

**Review by justin-richling — CHANGES_REQUESTED** [review]
@shawnusaf This all looks good, sorry for all the comments about commented code, if you could just go through and remove those lines I think we are good to merge this in!

File: lib/plotting_functions.py (hunk 2 above)
Comment (shawnusaf) [reply lib/plotting_functions.py]: Done.

File: scripts/plotting/global_latlon_map.py (hunk 12 above)
Comment (shawnusaf) [reply scripts/plotting/global_latlon_map.py]: Done

File: lib/adf_variable_defaults.yaml
Comment (shawnusaf) [reply lib/adf_variable_defaults.yaml:1]: Done

File: lib/plotting_functions.py (hunk 3 above)
Comment (shawnusaf) [reply lib/plotting_functions.py]: Done

File: lib/plotting_functions.py (hunk 4 above)
Comment (shawnusaf) [reply lib/plotting_functions.py]: Done

File: scripts/plotting/global_latlon_map.py (hunk 13 above)
Comment (shawnusaf) [reply scripts/plotting/global_latlon_map.py]: Done

File: scripts/plotting/polar_map.py (hunk 14 above)
Comment (shawnusaf) [reply scripts/plotting/polar_map.py]: Done

**Comment by shawnusaf** [comment]
I think I should have all of the commented lines removed.

**Review by justin-richling — APPROVED** [review]

**Pull request merged**